### DISCUSSION TITLE
Creating a "grid" experiment

### DISCUSSION THREAD

**Discussion by kimelit** [jspsych/jsPsych#810] [Q&A] [answered]
Hello,

Thanks so much for the course!

I want to create an experiment that looks like a grid: squares that I can press on each of them to change its color/play a wav file/change the text label or jpg displayed on this square. Where should I start? Is there a package in jsPsych that could be useful for that?

Looking forward to hearing from you,
Thank you!

**Comment by conbainbridge** [comment, marked as answer]
The plugin I would use as a starting point to build a new plugin like this would depend on which of those features (if not all) was going to be used. If you are intending to have audio play, I think the audio-button-response plugin could be a good starting point, as it already has audio setup as well the buttons that you can then customize.

Then, I would explore [Event Listeners](https://www.w3schools.com/js/js_htmldom_eventlistener.asp) for adding in the functionality you want. You will also want to move the existing click Event Listener, as it currently is used to advance to the next trial (in addition to logging the response). You would still need to add an additional button specifically for continuing to the next trial unless you want the plugin to be limited to advancing after a specified time only.

**Reply by kimelit** [reply]
@conbainbridge thanks! As of now I'm concerned with the display, that is how to create the visualization...

**Reply by conbainbridge** [reply]
This could be done in a few ways:

- If you want a grid that has a very strict layout (e.g. squares in a 5 x 5 layout), you could manually build it with a [html table](https://www.w3schools.com/html/html_tables.asp) into your layout. If you want to get fancy with it, you can design it to add rows/columns based on the number of choices (by finding the length of the choices away using trial.choices.length)
- You could also try [CSS grid layout](https://www.w3schools.com/css/css_grid.asp)
- For a responsive grid (blocks may change rows to accommodate changes in window size/dimensions, you could use [CSS float](https://www.w3schools.com/css/css_float.asp)

Hope this helps!

**Reply by kimelit** [reply]
ok, thanks..!
